src/js/app/baseimpl.ts

```diff
@@ -0,0 +1,39 @@
+import Base from "./base";
+import * as types from '../typings';
+import AssetId from "../utils/assetid";
+import { deleteObject, patchObject, postObject, getObjects } from "../api/datasets";
+
+/**
+ * When extending this class make sure to:
+ * a) set the static variable _type to the object type being implemented (e.g. space, file, bot etc)
+ *    this variable will be used to construct the endpoint URLs (NOTE: use singular)
+ * b) a static method _fromApi that processes the API payload and returns an array of objects
+ */
+export default abstract class BaseImpl extends Base {
+    static _type : string = 'unknown';
+    static _fromApi(payload: types.IAPIPayload) : Promise<Array<BaseImpl>> {
+        return Promise.reject('_newObj not implemented');
+    }
+    
+    static load(aid: AssetId) : Promise<Array<BaseImpl>> {
```

Comment:
`AssetId` is the correct thing to use here, however it breaks compatibility with the original functions with same name ... :/